From 5f467769cc4aa8d7dc0b775b0bce99d8cb466dff Mon Sep 17 00:00:00 2001
From: Richard Liu <r66033@freescale.com>
Date: Thu, 29 Aug 2013 16:42:06 +0800
Subject: [PATCH 469/486] ENGR00277268 enable hwcomposer version 1.2 on
 mx6sl_evk

enable hwcomposer version 1.2 on mx6sl_evk,
so that framework will pass plane alpha to
hwcomposer and use hwcomposer to do compostion
for alpha layer, that can improve performance.

Signed-off-by: Richard Liu <r66033@freescale.com>
---
 mx6/hwcomposer/Android.mk     | 3 +++
 mx6/hwcomposer/hwcomposer.cpp | 4 ++++
 2 files changed, 7 insertions(+)

diff --git a/mx6/hwcomposer/Android.mk b/mx6/hwcomposer/Android.mk
index 6cee365..af9aee1 100755
--- a/mx6/hwcomposer/Android.mk
+++ b/mx6/hwcomposer/Android.mk
@@ -41,6 +41,9 @@ LOCAL_MODULE := hwcomposer.$(TARGET_BOARD_PLATFORM)
 LOCAL_C_INCLUDES += hardware/imx/mx6/libgralloc_wrapper
 LOCAL_CFLAGS:= -DLOG_TAG=\"hwcomposer\"
 LOCAL_CFLAGS += -DENABLE_VSYNC
+ifneq ($(HAVE_FSL_IMX_GPU3D),true)
+LOCAL_CFLAGS += -DUSE_HWCOMPOSER_VERSION_1_2
+endif
 LOCAL_MODULE_TAGS := optional
 include $(BUILD_SHARED_LIBRARY)
 endif
diff --git a/mx6/hwcomposer/hwcomposer.cpp b/mx6/hwcomposer/hwcomposer.cpp
index 498597d..5fd45f0 100755
--- a/mx6/hwcomposer/hwcomposer.cpp
+++ b/mx6/hwcomposer/hwcomposer.cpp
@@ -324,7 +324,11 @@ static int hwc_device_open(const struct hw_module_t* module, const char* name,
 
         dev->device.prepare = hwc_prepare;
         dev->device.set = hwc_set;
+#ifdef  USE_HWCOMPOSER_VERSION_1_2
+        dev->device.common.version = HWC_DEVICE_API_VERSION_1_2;
+#else
         dev->device.common.version = HWC_DEVICE_API_VERSION_1_1;
+#endif
         dev->device.registerProcs = hwc_registerProcs;
         dev->device.eventControl = hwc_eventControl;
         dev->device.query = hwc_query;
-- 
1.8.0

